sources/packages/api/src/cra-integration/cra-files/cra-response-record-identification.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  TransactionCodes,
+  TransactionSubCodes,
+} from "../cra-integration.models";
+
+/**
+ * Common identification info for a CRA response file record.
+ * Please note that the numbers below (e.g. line.substr(0, 4)) represents
+ * the position of the information in a fixed text file format.
+ * The documentation about it is available on the document
+ * 'Income Verification Data Exchange Technical Guide BC'.
+ */
+export class CRAResponseRecordIdentification {
+  constructor(line: string, lineNumber: number) {
+    this.transactionCode = line.substr(0, 4) as TransactionCodes;
```

Comment:
IMO, instead of doing the substring logic inside the constructor we can have a simple constructor and and return transactionCode sin and transactionSubCode as getter methods. 

like this 

constructor(line: string, lineNumber: number) {
    this.line = line;
    this.lineNumber = lineNumber;
}

private readonly transactionCode: TransactionCodes;
  /**
   * SIN value that is part of the common record identification.
   */
  private readonly sin: string;
  /**
   * Secondary codes used alongside the records.
   */
  private readonly transactionSubCode: TransactionSubCodes;
  /**
   * Original line read from the CRA response file.
   */
  protected readonly line: string;
  /**
   * Original line number where this record was present
   * in the CRA response file. Useful for log.
   */
  protected readonly lineNumber: number;

public getSIN(){
 return this.line.substr(4, 9);
}

public getTransactionCode(){
 return this.line.substr(0, 4) as TransactionCodes;
}

Reply:
We have both ways in different files. Maybe we can do it all in one way or another.
I would prefer to keep this logic in the constructor because moving it to the get method will make the _substr_ method be executed every time that the property is accessed, does it make sense?
The only benefit, IMO, to defer the execution of the substring to the get method would be if the logic in the constructor is heavy and must be deferred or if the property would not likely be accessed.
Please let me know your thoughts about it.

Reply:
Technically what you said makes sense. But practically would we call getSIN() more than once? because once it is called we assign it to const or map it to another object for transformation or else evaluate an expression.  So multiple calls to getSIN or any other getter is not practically required. 

There is no major difference between both the ways. So any of them should be fine.